server/tests/service_discovery_test.py

```diff
@@ -0,0 +1,41 @@
+import tempfile
+import unittest
+
+from services.discovery import configure_endpoints_from_ingress
+from services.discovery import get_service_url
+
+
+class TestServiceDiscovery(unittest.TestCase):
+
+  def test_configure_endpoints_from_ingress(self):
+    ingress_rules = '''query-host:8080:
```

Comment:
maybe use a yaml file instead of inline string (easier to view and update) with complete coverage of all the rules

Reply:
done.